syntax = "proto3";

package translate;

import "google/protobuf/empty.proto";

service TranslateService {
  rpc LoadModel (google.protobuf.Empty) returns (google.protobuf.Empty) {}
  rpc FreeModel (google.protobuf.Empty) returns (google.protobuf.Empty) {}
  rpc Translate (TranslateRequest) returns (stream TranslateResponse) {}

  rpc WatchLog (google.protobuf.Empty) returns (stream LogResponse) {}
}

message TranslateRequest {
  string content = 1;
  bool is_jp_to_en = 2;
}

message TranslateResponse {
  string content = 1;
}

message LogResponse {
  string log_content = 1;
}